arroyo/processing/strategies/run_task_with_multiprocessing.py

```diff
@@ -604,12 +646,54 @@ def __check_for_results_impl(self, timeout: Optional[float] = None) -> None:
                         result.next_index_to_process,
                     ),
                 ),
+                input_block_too_small,
+                True,
             )
             return
 
+        old_input_block = input_batch.block
+
+        if (
+            input_block_too_small
+            and self.__resize_input_blocks
+            and (
+                self.__max_input_block_size is None
+                or self.__max_input_block_size > old_input_block.size * 2
+            )
+        ):
+            self.__metrics.increment(
+                "arroyo.strategies.run_task_with_multiprocessing.batch.input.resize"
+            )
+            new_input_block = self.__shared_memory_manager.SharedMemory(
+                old_input_block.size * 2
```

Comment:
This seems like it could trigger quite a few resizes depending on the values passed. Is there another way to calculate the sharedmemory size based on prior seen message sizes?

Reply:
I think that's a good idea, we can keep track of the batch size and use it as input for reallocating. I'll have to see how to do that.. I don't think I can use our existing batch size metrics because they don't emit the right value when batches are split up due to input/output overflow.